go/rpcclientlib/obscuro_client.go

```diff
@@ -177,10 +177,14 @@ func (c *networkClient) decryptResponse(resultBlob interface{}) ([]byte, error)
 		// todo: remove this when we no longer support disabling viewing keys for testing
 		return unencrypted, nil
 	}
-	encryptedResult := common.Hex2Bytes(resultBlob.(string))
+	resultStr, ok := resultBlob.(string)
+	if !ok {
+		return nil, fmt.Errorf("expected hex string but result was not a string, it was (%t) %s", resultBlob, resultBlob)
+	}
+	encryptedResult := common.Hex2Bytes(resultStr)
 	decryptedResult, err := c.viewingPrivKey.Decrypt(encryptedResult, nil, nil)
 	if err != nil {
-		return nil, fmt.Errorf("could not decrypt enclave response with viewing key: %w", err)
+		return nil, fmt.Errorf("could not decrypt response: %s, with viewing key, error: %w", resultStr, err)
```

Comment:
I think this might be clearer as `fmt.Errorf("could not decrypt the following response with viewing key: %s. Cause: %w", resultStr, err)`